### ISSUE TITLE
Wrong/outdated code snippets in the docs

### ISSUE THREAD

**Issue by millenc** [dmishh/SettingsBundle#82]
I've just begun using this bundle and I've come across a few code snippets in the documentation that are wrong or outdated when compared to the current master. Proposed fixes in the pull request  #83:

### Resources/doc/installation.md

master:

```php
<?php

// in AppKernel::registerBundles()
$bundles = array(
    // ...
    new Dmishh\SettingsBundle\DmishhSettingsBundle(),
);
```
is missing the *Bundle* part of the namespace just before *SettingsBundle*.

Fixed:

```php
<?php

// in AppKernel::registerBundles()
$bundles = array(
    // ...
    new Dmishh\Bundle\SettingsBundle\DmishhSettingsBundle(),
);
```

### Resources/doc/advanced-configuration.md

master:

```yaml
dmishh_settings:
    template: DmishhSettingsBundle:Settings:manage.html.twig
    cache_service: null
    cache_lifetime: 3600
    security:
         manage_global_settings_role: ROLE_USER
         users_can_manage_own_settings: true
    serialization: php # database serialization mechanism (php|json)
    settings:
        my_first_setting:
            type: number # any Symfony form type
            options: # options passed to form
                required: false
            constraints:
                Symfony\Component\Validator\Constraints\Range:
                    min: 1
                    max: 65535
```
The *options* and *constraints* sections of *my_first_setting* appear directly below it but they should go below a *validation* section. Also the *constraints* section must go under the *options* section.

Fixed:

```yaml
dmishh_settings:
    template: DmishhSettingsBundle:Settings:manage.html.twig
    cache_service: null
    cache_lifetime: 3600
    security:
         manage_global_settings_role: ROLE_USER
         users_can_manage_own_settings: true
    serialization: php # database serialization mechanism (php|json)
    settings:
        my_first_setting:
        	validation:
	            type: number # any Symfony form type
	            options: # options passed to form
	                required: false
		            constraints:
		                Symfony\Component\Validator\Constraints\Range:
		                    min: 1
		                    max: 65535
```

master:

```yaml
dmishh_settings:
    settings:
        my_first_setting:
            type: text
            options:
                max_length: 15
            constraints:
                Symfony\Component\Validator\Constraints\Regex:
                    pattern: "/^\d+$/"
```

Fixed:

```yaml
dmishh_settings:
    settings:
        my_first_setting:
            validation:
                type: text
                options:
                    max_length: 15
                    constraints:
                        Symfony\Component\Validator\Constraints\Regex:
                            pattern: "/^\d+$/"
```

### Resources/doc/faq.md

master:

```yaml
dmishh_settings:
    settings:
        my_first_setting:
            options:
                required: false
```
same error as **Resources/doc/advanced-configuration.md**

fixed:

```yaml
dmishh_settings:
    settings:
        my_first_setting:
        	validation:
	            options:
	                required: false
```

**Comment by rvanlaak** [comment]
Hi @millenc, both your proposed fixes reflect changes we made lately to simplify the bundle. Probably you don't have `master` or any of the `2.0.0` betas installed, but do you have a pre-2.0 version.

The namespace changed, as you can see in the class [DmishhSettingsBundle](https://github.com/dmishh/SettingsBundle/blob/master/DmishhSettingsBundle.php#L12)

Concerning the validation, we chose to make the configuration less deep. That was modified in https://github.com/dmishh/SettingsBundle/commit/23f89327b8720008fde94ee96619bb73eff7da82

Can you validate this by installing the latest beta release?

**Comment by millenc** [comment]
Sorry @rvanlaak, my bad. You are absolutely right. I installed the package using the instructions on the Installation section of the docs (adding the dependency ""dmishh/settings-bundle": "2.0.*@dev"" to my composer.json file). This makes Composer download the 2.0.0-beta1 version instead of the latest beta3 (I don't know if this is intended or an issue with my composer installation, which is pretty old as far as I know). This version does have the namespace change but not the changes related to the validation section of the configuration (those were introduced in the 2.0.0-beta2 version).

My mistake was reading the docs on master and assuming that it was the version that I was using without actually double-checking it. Sorry for any inconveniences I may have caused. Feel free to close this issue and the corresponding pull request as non relevant.

**Comment by rvanlaak** [comment]
No problem at all, maybe you've installed `beta1` before and is it set to that version via `composer.lock`. Can you try to do `composer update`?

**Comment by millenc** [comment]
Yeah, I did try to do `composer update` but the result was the same. I've also tried updating Composer and running `composer clearcache` before `update` to make sure it's not a caching issue. None of this worked.

Fortunately I've discovered the issue. I've tried to install the exact `2.0.0-beta3` version and composer complains about not meeting the platform requirements (PHP version). I'm using PHP 5.6.15 but my composer.json for some reason has a `config.platform.php` setting with value `5.3.9`. I don't remember if I installed Symfony through the command line tool or by downloading the standard distribution zip, but this setting was already there from the beginning. This setting makes composer simulate that I'm using that particular PHP version instead of the actual one (PHP 5.6.15). Starting with the `2.0.0-beta2` version the bundle requires `"php": "^5.5.9|^7.0",` , so I was stuck with the `2.0.0-beta1` version which requires `"php": ">=5.3.3",`.

I've removed these lines:
```
"platform": {
     "php": "5.3.9"
}
```
from the `config` section of my composer.json and doing `composer update` now makes composer download the latest version. I think that in my case this configuration makes no sense so I'm leaving those lines out in order to be able to use the latest version of the bundle.

Thank you very much for your support @rvanlaak!
